corgi/api/filters.py

```diff
@@ -37,11 +37,17 @@ def filter(self, queryset: QuerySet, value: str) -> QuerySet:
         for tag in search_tags:
             if ":" in tag:
                 tag_name, _, tag_value = tag.partition(":")
-                queryset = queryset.filter(
-                    tags__name__icontains=tag_name, tags__value__icontains=tag_value
-                )
+                if tag_name.startswith("!"):
+                    tag_name = tag_name[1:]
+                    queryset = queryset.exclude(tags__name=tag_name, tags__value=tag_value)
+                else:
+                    queryset = queryset.filter(tags__name=tag_name, tags__value=tag_value)
             else:
-                queryset = queryset.filter(tags__name__icontains=tag)
+                if tag.startswith("!"):
+                    tag = tag[1:]
+                    queryset = queryset.exclude(tags__name=tag)
+                else:
+                    queryset = queryset.filter(tags__name=tag)
```

Comment:
We went from doing a case-insensitive substring match (using explicit __icontains) to doing a case-sensitive exact match (using implicit __exact / no explicit lookup given). LGTM if that's expected.

Minor suggestion: below can be simplified a bit so we're not repeating code four different ways:
```suggestion
            exclude = False
            if tag.startswith("!"):
                tag = tag[1:]
                exclude = True
            if ":" in tag:
                tag_name, _, tag_value = tag.partition(":")
                queryset_kwargs = {"tags__name": tag_name, "tags__value": tag_value}
            else:
                queryset_kwargs = {"tags__name": tag}
            if exclude:
                queryset = queryset.exclude(**queryset_kwargs)
            else:
                queryset = queryset.filter(**queryset_kwargs)
```

Reply:
ok, updated. Yes, I meant to remove the case-insensitive sub string match. Since we will soon have a way to enumerating tags, we should be able to discover tags that way.